ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -84,6 +87,13 @@ public void setRenderToHardwareTexture(T view, boolean useHWTexture) {
 
   @ReactProp(name = PROP_TEST_ID)
   public void setTestId(T view, String testId) {
+    if (!testIDs.containsKey(testId)) {
```

Comment:
`view.getResources().getIdentifier(testId, "id", view.getContext().getPackageName())` will return `0` if no matching identifier is found.  I wanted to avoid calling it more than once for a given testID if possible which is why there are 2 check:

* The first check is for caching purposes.
* The second check is to avoid setting the testId on the View if the user didn't define it.